migrating_gazebo_classic_ros2_packages.md

```diff
@@ -0,0 +1,593 @@
+  <max_linear_acceleration>0.033</max_linear_acceleration> <!-- computed from <max_linear_acceleration> in the original plugin-->
+
+  <topic>cmd_vel</topic> <!-- from <commant_topic> -->
+
+  <odom_topic>odom</odom_topic> <!-- from <odometry_topic> -->
+  <frame_id>odom</frame_id> <!-- from <odometry_frame> -->
+  <child_frame_id>base_footprint</child_frame_id> <!-- from <robot_base_frame> -->
+  <odom_publisher_frequency>30</odom_publisher_frequency> <!-- from <update_rate>-->
+
+  <tf_topic>/tf</tf_topic> <!-- Short topic name for tf output -->
+
+</plugin>
+
+```
+
+The `<wheel_torque>` parameter can be realized by setting effort limits on each
+`<joint>`. For example:
+
+```xml
+
+<joint name="wheel_right_joint" type="revolute">
+  <parent>base_link</parent>
+  <child>wheel_right_link</child>
+  <pose>0.0 -0.144 0.023 -1.57 0 0</pose>
+  <axis>
+    <xyz>0 0 1</xyz>
+    <limit>
+      <effort>20</effort> <!-- from <wheel_torque> in libgazebo_ros_diff_drive.so.-->
+    </limit>
+  </axis>
+</joint>
+```
+
+### libgazebo_ros_joint_state_publisher.so
+
+We will replace this plugin as well with
+[`JointStatePublisher`](https://gazebosim.org/api/gazebo/6/classignition_1_1gazebo_1_1systems_1_1JointStatePublisher.html).
+The parameters are mostly similar, however, the `<update_rate>` parameter is not
+supported. Here's the full `<plugin>` tag with comments describing the mapping
+from the original plugin.
+
+```xml
+<plugin filename="gz-sim-joint-state-publisher-system"
+  name="gz::sim::systems::JointStatePublisher">
+  <topic>joint_states</topic> <!--from <ros><remaping> -->
+  <joint_name>wheel_left_joint</joint_name>
+  <joint_name>wheel_right_joint</joint_name>
+</plugin>
+```
+
+### World plugins
+
+As mentioned earlier, sensors are handled by generic world level plugins. By
+default, if there are no world plugins specified, Gazebo adds the `Physics`,
+`SceneBroadcaster`, and `UserCommands` plugins by default. However, if we
+specify any plugins at all, Gazebo will assume we want to override the defaults,
+so will not add any default plugins. Therefore, we have to add the additional
+plugins for IMU and Lidar sensors as well as the ones that would have been added
+by default. We will once again edit `turtlebot3_gazebo/worlds/empty_world.world`
+add the following right after `<world name="default">`.
+
+```xml
+<plugin
+  filename="gz-sim-physics-system"
+  name="gz::sim::systems::Physics">
+</plugin>
+<plugin
+  filename="gz-sim-user-commands-system"
+  name="gz::sim::systems::UserCommands">
+</plugin>
+<plugin
+  filename="gz-sim-scene-broadcaster-system"
+  name="gz::sim::systems::SceneBroadcaster">
+</plugin>
+<plugin
+  filename="gz-sim-sensors-system"
+  name="gz::sim::systems::Sensors">
+  <render_engine>ogre2</render_engine>
+</plugin>
+<plugin
+  filename="gz-sim-imu-system"
+  name="gz::sim::systems::Imu">
+</plugin>
+```
+
+## Bridge ROS topics
+
+The plugins in `gazebo_ros_pkgs` are Gazebo plugins that are also ROS nodes. In
```

Comment:
Can we reword this just a little bit to make it clearer? I can't make suggestions on multi-line text, but I would make it clear Gazebo Classic, when using gazebo_ros_pkgs uses a ROS node per plugin while New Gazebo working with ros_gz bridges individual topics. Currently we refer to just gazebo_ros_pkgs and ros_gz, and while that's correct it makes the reader have to map between those package and old and new Gazebo.

Reply:
Reworded in e39cabe74c089f5ada56de62f660c73ac203263a